src/SignalR/clients/java/signalr/test/src/main/java/com/microsoft/signalr/HubConnection.ReturnResultTest.java

```diff
@@ -0,0 +1,100 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+package com.microsoft.signalr;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import java.nio.ByteBuffer;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import io.reactivex.rxjava3.subjects.SingleSubject;
+
+@ExtendWith({RxJavaUnhandledExceptionsExtensions.class})
```

Comment:
Test with an async result?

Reply:
> Ignore most of the code right now

Yeah tests need to be written, and need to figure out how to properly wait for the results internally.